File: leetcode/array/624. Maximum Distance in Arrays/py/main.py
class Solution:
    def maxDistance(self, arrays):
      if len(arrays) == 0:
        return 0

      max_val = arrays[0][-1]
      min_val = arrays[0][0]
      res = 0

      for i in range(1, len(arrays)):
        res = max(res, abs(arrays[i][-1] - min_val), abs(arrays[i][0] - max_val))
        min_val = min(min_val, arrays[i][0])
        max_val = max(max_val, arrays[i][-1])
        
      return res

File: leetcode/array/624. Maximum Distance in Arrays/js/main.js
// Time O(N)
// Space O(1)
const maxDistance = arrays => {
  if (arrays.length === 0) return 0;

  let ans = 0;
  let min = arrays[0][0];
  let max = arrays[0][arrays[0].length - 1];

  for (let i = 1; i < arrays.length; i++) {
    ans = Math.max(ans, Math.abs(arrays[i][arrays[i].length - 1] - min), Math.abs(arrays[i][0] - max));
    min = Math.min(min, arrays[i][0]);
    max = Math.max(max, arrays[i][arrays[i].length - 1]);
  }

  return ans;
};
